numpyro/infer/elbo.py

```diff
@@ -370,6 +370,16 @@ def _single_particle_elbo(self, model, guide, param_map, args, kwargs, rng_key):
             indep_plates = set.intersection(*site_plates.values())
         else:
             indep_plates = set()
+        for frame in set.union(*site_plates.values()):
+            if frame not in indep_plates:
+                subsample_size = frame.size
+                size = model_trace[frame.name]["args"][0]
+                if size > subsample_size:
+                    raise ValueError(
+                        f"Subsample plate `{frame.name}` should cover"
```

Comment:
how about

`RenyiELBO` only supports subsampling in plates that are common to all `sample` sites, e.g. a data plate that encloses the entire model.

Reply:
done